addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -26,12 +26,16 @@ def _compute_is_forward_operator(self):
         self.filtered(lambda s: s.step_type == "create_lead_and_forward").is_forward_operator = True
 
     def _chatbot_crm_prepare_lead_values(self, discuss_channel, description):
+        partner = self.env.user.partner_id
+        team = self.crm_team_id
+        if partner.company_id and team.company_id and partner.company_id != team.company_id:
+            team = self.env["crm.team"]
         return {
-            "company_id": self.crm_team_id.company_id.id,
+            "company_id": team.company_id.id or partner.company_id.id,
             'description': description + discuss_channel._get_channel_history(),
             "name": self.env._("%s's New Lead", self.chatbot_script_id.title),
             'source_id': self.chatbot_script_id.source_id.id,
-            'team_id': self.crm_team_id.id,
+            "team_id": team.id,
```

Comment:
You should change all `self.crm_team_id` to `team` or none of them.

Reply:
I think we should actually not assign a `type` if no team is selected.